src/ucp/core/ucp_mm.c

```diff
@@ -1774,37 +1774,38 @@ ucp_memh_import_attach(ucp_context_h context, ucp_mem_h memh,
     uct_mem_h uct_memh;
 
     for (tl_mkey_index = 0; tl_mkey_index < num_tl_mkeys; ++tl_mkey_index) {
-        md_index    = tl_mkeys[tl_mkey_index].md_index;
-        tl_mkey_buf = tl_mkeys[tl_mkey_index].tl_mkey_buf;
-        md_attr     = &context->tl_mds[md_index].attr;
-        ucs_assert_always(md_attr->flags & UCT_MD_FLAG_EXPORTED_MKEY);
+        ucs_for_each_bit(md_index, tl_mkeys[tl_mkey_index].local_md_map) {
+            tl_mkey_buf = tl_mkeys[tl_mkey_index].tl_mkey_buf;
```

Comment:
minor: we can initialize it outside the ucs_for_each_bit loop because it depends only on tl_mkey_index

Reply:
sure